Nondeterministic test failure

This test fails nondeterministically on CI:

test_generators.py::test_inpainting_generator_random_ratio[dtype0-cpu-3]

**Failing CI**

https://github.com/deepinv/deepinv/actions/runs/15652961332/job/44100164612

**Stack trace**

```
    @pytest.mark.parametrize("num_channels", NUM_CHANNELS)
    @pytest.mark.parametrize("device", DEVICES)
    @pytest.mark.parametrize("dtype", DTYPES)
    def test_inpainting_generator_random_ratio(num_channels, device, dtype):
        # NOTE elements of this test are now redundant given above tests
        size = (100, 100)  # we take it large to have significant statistical numbers after
        physics = dinv.physics.Inpainting((num_channels, size[0], size[1]), 0.9)
    
        split_ratio = 0.6
        generator = dinv.physics.generator.BernoulliSplittingMaskGenerator(
            (num_channels, size[0], size[1]), split_ratio=split_ratio
        )
        batch_size = 2
        params = generator.step(batch_size=batch_size)
    
        mask = params["mask"]
        assert mask.shape == (batch_size, num_channels, size[0], size[1])
    
        experimental_split_ratio = (mask[0] == 1).sum() / mask[0].numel()
        assert abs(experimental_split_ratio.item() - split_ratio) < 1e-2
    
        # check forward
        x = torch.randn((batch_size, num_channels, size[0], size[1]))
        y = physics(x, **params)
        experimental_split_ratio_obs = 1 - (y[0] == 0).sum() / y[0].numel()
>       assert experimental_split_ratio == experimental_split_ratio_obs
E       assert tensor(0.6016) == tensor(0.6016)

deepinv/tests/test_generators.py:485: AssertionError
```

**Fix**

The test should be made deterministic by controlling the RNG.

--- a/deepinv/tests/test_generators.py
+++ b/deepinv/tests/test_generators.py
@@ -459,6 +459,8 @@ def test_inpainting_generators(
 @pytest.mark.parametrize("device", DEVICES)
 @pytest.mark.parametrize("dtype", DTYPES)
 def test_inpainting_generator_random_ratio(num_channels, device, dtype):
+    rng = torch.Generator().manual_seed(0)
+
     # NOTE elements of this test are now redundant given above tests
     size = (100, 100)  # we take it large to have significant statistical numbers after
     physics = dinv.physics.Inpainting((num_channels, size[0], size[1]), 0.9)
@@ -477,10 +479,12 @@ def test_inpainting_generator_random_ratio(num_channels, device, dtype):
     assert abs(experimental_split_ratio.item() - split_ratio) < 1e-2
 
     # check forward
-    x = torch.randn((batch_size, num_channels, size[0], size[1]))
+    x = torch.randn((batch_size, num_channels, size[0], size[1]), generator=rng)
     y = physics(x, **params)
     experimental_split_ratio_obs = 1 - (y[0] == 0).sum() / y[0].numel()
-    assert experimental_split_ratio == experimental_split_ratio_obs
+    assert torch.allclose(
+        experimental_split_ratio, experimental_split_ratio_obs, rtol=1e-4
+    )
 
     # now we do the same with each element in the batch for random_split_ratio
     min_split_ratio = 0.001
@@ -498,7 +502,7 @@ def test_inpainting_generator_random_ratio(num_channels, device, dtype):
     mask = params["mask"]
     assert mask.shape == (batch_size, num_channels, size[0], size[1])
 
-    x = torch.randn((batch_size, num_channels, size[0], size[1]))
+    x = torch.randn((batch_size, num_channels, size[0], size[1]), generator=rng)
     y = physics(x, **params)
 
     list_exp_split_ratio = []

--- a/deepinv/tests/test_generators.py
+++ b/deepinv/tests/test_generators.py
@@ -458,9 +458,8 @@ def test_inpainting_generators(
 @pytest.mark.parametrize("num_channels", NUM_CHANNELS)
 @pytest.mark.parametrize("device", DEVICES)
 @pytest.mark.parametrize("dtype", DTYPES)
-def test_inpainting_generator_random_ratio(num_channels, device, dtype):
-    rng = torch.Generator().manual_seed(0)
-
+@pytest.mark.parametrize("rng", [torch.Generator().manual_seed(0)])
+def test_inpainting_generator_random_ratio(num_channels, device, dtype, rng):
     # NOTE elements of this test are now redundant given above tests
     size = (100, 100)  # we take it large to have significant statistical numbers after
     physics = dinv.physics.Inpainting((num_channels, size[0], size[1]), 0.9)

